### ISSUE TITLE
Increase sourceCompatibility to Java 11

### ISSUE THREAD

**Issue by abelsromero** [spring-io/concourse-release-scripts#20]
Given the tool is aimed to be run in a docker image, is there any reason to keep it at 8?

**Comment by bclozel** [comment]
Indeed, this is meant to be used in a docker image, but right now several projects are embedding this tool in their main CI image - there, the default JDK version available is aligned with the minimum requirements for Spring projects (JDK8, as of this writing).

It would be easier to upgrade to JDK11 here if we were shipping a proper Concourse resource (so the docker image itself), rather than a JAR.

With that in mind, I'm tempted to keep things as is and only upgrade to the minimum JDK requirementfor Spring projects in the future. Is there a critical aspect that I'm missing here?
